packages/agents/lighthouse/src/tasks/prover/operations/process.ts

```diff
@@ -207,7 +207,17 @@ export const processMessages = async (brokerMessage: BrokerMessage, _requestCont
       requestContext,
     );
     logger.info("Proved and processed message sent to relayer", requestContext, methodContext, { taskId });
+    if (taskId) {
+      await cache.messages.removePending(
+        originDomain,
+        destinationDomain,
+        messages.map((it) => it.leaf),
```

Comment:
should only remove the message from pending, if a proof was included in this batch.

Reply:
Fixed